tools/explorer/lib/ontology_wrapper.py

```diff
@@ -105,58 +103,225 @@ def GetFieldsForTypeName(
+    belonging to an entity type and concrete entity to create a big string
+    representation of matching fields within a match. This method will be called
+    by app.py when a user wants to visualize the field relationships between a
+    list of fields and an entity type within a match.
+
+    Args:
+      match: An instance of Match class for which a field set relation wants
+      to be visualized.
     Returns:
-          entities: a list of EntityType objects matching the provided list of
-          fields.
+      A string to visualize a field set relation
     """
-    pass
+    final_matrix = []
+    concrete_field_set = set(match.GetFieldList())
+
+    canonical_field_dict = {}
+    for qualified_field in match.GetEntityType().GetAllFields().values():
+      new_entity_type_field = EntityTypeField(
+          qualified_field.field.namespace,
+          qualified_field.field.field,
+          qualified_field.optional,
+          qualified_field.field.increment
+      )
+      new_standard_field = model.StandardizeField(new_entity_type_field)
+      canonical_field_dict[new_standard_field] = new_entity_type_field
+
+    standard_canonical_field_set = set(canonical_field_dict.keys())
+
+    only_concrete = concrete_field_set.difference(standard_canonical_field_set)
+    for field in only_concrete:
+      final_matrix.append([str(field), '', ''])
+
+    intersection = standard_canonical_field_set.intersection(concrete_field_set)
+    for field in intersection:
+      final_matrix.append(
+          [str(field), str(field), canonical_field_dict[field].IsOptional()])
+
+    only_canonical = standard_canonical_field_set.difference(concrete_field_set)
+    for field in only_canonical:
+      final_matrix.append(
+          ['', str(field), canonical_field_dict[field].IsOptional()])
+
+    padding = 3
+    all_fields = list(intersection) + list(only_concrete) + list(only_canonical)
+    col_width = max(len(str(field)) for field in all_fields) + padding
+    return_string = ''
+    return_string += colored('MATCH SCORE: ', 'yellow')
+    return_string += str(match.GetMatchScore()) + '\n'
+    return_string += colored('MATCHED TYPE: ', 'yellow')
+    return_string += str(match.GetEntityType().typename) + '\n'
+    return_string += '\n'
+    return_string += ''.join(
+        colored(field.ljust(col_width), 'yellow')
+        for field in ['ACTUAL FIELDS', 'TYPE FIELDS', 'OPTIONALITY'])
+    return_string += '\n'
+    return_string += ''.join(
+        field.ljust(col_width) for field in ['='*(col_width-padding)]*3)
+    return_string += '\n'
+    for row in final_matrix:
+      if not row[2]:
+        row[2] = 'Required'
+      elif row[2]:
+        row[2] = 'Optional'
+      elif row[2] == '':
+        continue
+      if row[0] != '' and row[1] != '':
+        return_string += ''.join(
+            colored(field.ljust(col_width), 'green') for field in row)
+      else:
+        return_string += ''.join(field.ljust(col_width) for field in row)
```

Comment:
would it make sense to refactor this part to be used later? this method is almost a 100 lines

Reply:
Can we leave it for now? It's just a lot of formatting.

Reply:
There's also nothing to really break apart. It's just creating one big string block.

Reply:
suggested a populatMatrix above